src/math/webgl/shader_compiler.ts

```diff
@@ -130,7 +130,9 @@ vec2 UVfrom2D(int texNumR, int texNumC, int numC, int row, int col) {
 const SAMPLE_3D_SNIPPET = `
 vec2 UVfrom3D(int texNumR, int texNumC, int stride0,
     int stride1, int row, int col, int depth) {
-  int index = row * stride0 + col * stride1 + depth;
+  vec3 v1 = vec3(row, col, depth);
```

Comment:
Unfortunately, dot() works only on floats, and for large `ndarray`s, the flat index can be high (>10 million), at which point float is loosing precision and causes index to be `10,000,001` instead of `10,000,000` for example. We had a real model affected by this.